site/docs/v1/tech/lambdas/self-service-registration.adoc

```diff
@@ -0,0 +1,70 @@
+---
+layout: doc
+title: Self Service Registration Validation
+description: An overview of the self-service registration validation lambda
+navcategory: customization
+---
+
+:sectnumlevels: 0
+
+== Self Service Registration Validation lambda
+
+When you have an link:/docs/v1/tech/guides/advanced-registration-forms[Advanced Registration Form] you may add a lambda
+to perform additional validation of the form fields as the user completes steps in the form.
+
+When you create a new lambda using the FusionAuth UI we will provide you an empty function for you to implement. If you are using the API to create the lambda you will need to ensure your function has the following signature:
+
+[source,javascript]
+----
+function validate(result, user, registration, formContext) {
+  // Lambda code goes here
+}
+----
+
+This lambda must contain a function named `validate` that takes four parameters parameters. The parameters that the lambda is passed are:
+
+* `result` - this is a FormValidationResult object with one property
+  - `errors` which is a FusionAuth Errors object to append form errors to, and is the output of the function
+* `user` - the FusionAuth User object
+* `registration` - the FusionAuth UserRegistration object
+* `formContext` - an object containing data about the current form state. If has the following properties:
+  - `fields` - an array of FormFields in the current form step
+  - `form` - the FusionAuth Form object
+  - `step` - the current step number
+  - `stepIndex` - the zero-indexed number of the current step (always step - 1)
+  - `totalSteps` - the total number of steps in the form
+
+The `result` Errors object is documented in the link:/docs/v1/tech/apis/errors[Errors API]
```

Comment:
These documentation link lines should have a period at the end as they are all complete sentences.

Reply:
How does this read in HTML? I think it will generate a bunch of paragraphs with a single line. Would it be better as a bulleted list?

Reply:
<img width="1396" alt="Screenshot 2023-01-12 at 10 01 51 AM" src="https://user-images.githubusercontent.com/11809481/212132004-3011eceb-f81b-4ff7-ad4f-64e2b0d88162.png">

?

Reply:
Could these links be combined with the list of parameters above? For example:
```
* `user` - the FusionAuth User object documented in the link:/docs/v1/tech/apis/users[User API]
```